runtime/ts/manifest.ts

```diff
@@ -939,6 +939,26 @@ ${e.message}
           store.referenceMode = false;
           await store.set(hostedParticleLiteral);
           targetHandle.mapToStorage(store);
+        } else if (connection.type && connection.type.isInterface) {
+          let hostedParticleStore = manifest.findStoreById(targetHandle.id);
```

Comment:
is it possible to extract the type off the store directly? I don't like the idea of instantiating storage inside manifest.ts (e.g. it's possible that this is just a StorageStub to a firebase location and the .get() call below will fail).

If it isn't, then we might need to look at how the store types are computed in these cases and make that more restrictive.